core/src/main/scala/org/apache/spark/api/python/PythonRunner.scala

```diff
@@ -549,6 +549,13 @@ private[spark] abstract class BasePythonRunner[IN, OUT](
         JavaFiles.deleteIfExists(path)
         throw new SparkException(s"Python worker exited unexpectedly (crashed): $error", e)
 
+      case eof: EOFException if !faultHandlerEnabled =>
+        throw new SparkException(
+          s"Python worker exited unexpectedly (crashed). " +
+            "Consider setting 'spark.sql.execution.pyspark.udf.faulthandler.enabled' or" +
```

Comment:
https://github.com/apache/spark/blob/9a990b5a40de3c3a17801dd4dbd4f48e3f399815/core/src/main/scala/org/apache/spark/api/python/PythonRunner.scala#L112C1-L113C1

It seems that `faultHandlerEnabled` only checks `spark.python.worker.faulthandler.enabled`. So, if we only configure `spark.sql.execution.pyspark.udf.faulthandler.enabled` to be true, will it really enter the matching of `faultHandlerEnabled` being true?

Reply:
It's inherited :-).